components/ws-manager-mk2/controllers/status.go

```diff
@@ -123,6 +140,16 @@ func (r *WorkspaceReconciler) updateWorkspaceStatus(ctx context.Context, workspa
 		workspace.Status.Phase = *phase
 	}
 
+	if failure != "" && !workspace.IsConditionTrue(workspacev1.WorkspaceConditionPodRejected) {
+		// Check: A situation where we want to retry?
+		if isPodRejected(pod) {
```

Comment:
> The kubelet will also attempt to re-admit the pod upon kubelet restart

Ah, did not know, thanks for sharing!

My approach was that today, these cases end as "failed". So whatever I do, I prevent these from happening now, by trying to restart. If what you say is applicable, yes, there might be data loss - but so was before (yes, manual recovery is a thing, but not in Dedicated).

Reply:
> If the pod is simply marked as failed, it will continue to perform the standard backup operation.

You are right indeed. :dart: 

We can add that in a follow-up, though.

Reply:
Linear issue: https://linear.app/gitpod/issue/CLC-942/[ws-manager]-avoid-podrejection-on-kubelet-restart